### DISCUSSION TITLE
Does Umami provide a fallback if no JavaScript is enabled?

### DISCUSSION THREAD

**Discussion by AngelCrawford** [umami-software/umami#1022] [Q&A] [answered]
Does Umami provide a tracking pixel fallback, if the user has JavaScript deactivated in his Browser?

Greetings, 
Angel

**Comment by binyamin** [comment, marked as answer]
> The goal is to provide you with a friendlier, **privacy-focused** alternative to Google Analytics....

It's probably best to respect the user's wishes. If a user turns off JavaScript, they might not _want_ to be tracked. Also, there may be legal issue here.

### Legal Considerations
**Disclaimer**:  I'm not a lawyer. I just did some homework.

I don't think it's possible to do that legally.
1. A tracking pixel needs consent
    1. Like with any image, browsers will download the tracking pixel automatically.
    2. According to PECR (a UK law similar to GDPR), the user must give consent _before_ you store or collect information.[^pecr]
    3. Since a tracking pixel is downloaded, it needs consent.
2.  Because JavaScript is blocked, there's no way to wait for consent before downloading the tracking pixel.

[^pecr]: https://ico.org.uk/for-organisations/guide-to-pecr/guidance-on-the-use-of-cookies-and-similar-technologies/what-are-the-rules-on-cookies-and-similar-technologies#rules1

**Comment by AngelCrawford** [comment]
> Because JavaScript is blocked, there's no way to wait for consent before downloading the tracking pixel.

Oh, this was the fact that hit me, you are completely right. Didn't thought about this, I'm learning >.< 

Thank you :)
